codecs/README.md

```diff
@@ -0,0 +1,32 @@
+# Codecs
+
+This folder contains a self-contained sub-project for each codec that squoosh supports.
+
+## Build
+
+Each subproject can be built using the following commands:
+
+```
+$ npm install
+$ npm run build
+```
+
+This will build two files: `<codec name>_codec.js` and `<codec name>_codec.wasm`. Due to some current limitations in Emscripten, both of these files have to be served from the root folder. When the `.js` file is loaded, a global `<codec name>` is created with the same API as an [Emscripten `Module`](https://kripken.github.io/emscripten-site/docs/api_reference/module.html).
+
+Each codec offers at least 2 functions:
+
+- `void encode(uint8_t* img_in, int width, int height, float quality)`
```

Comment:
We're probably going to end up with very different calls for each codec, but I guess in, width, & height will be consistent.

Reply:
I thought we could keep it consistent across all codecs. I guess the options are gonna make that impossible?

Reply:
Yeah, I think the options make it impossible. Even `quality` doesn't make sense for lossless codecs.

Keeping the first 3 args in, width & height seems sensible though.

Reply:
At that point we’ll have to write a JS wrapper for the module anyways. 🤷‍♂️

Reply:
Yeah, I think we'll end up with something like that anyway to wrap up an array of output bytes along with a `.free()` method.